modules/@apostrophecms/doc/index.js

```diff
@@ -219,9 +219,12 @@ module.exports = {
         }
       },
       '@apostrophecms/doc-type:afterDelete': {
-        // If deleting draft also delete published, but
-        // not vice versa ("undo publish" is a thing).
-        async deleteOtherModeAfterDelete(req, doc, options) {
+        // Deleting a draft implies deleting the document completely, since
+        // a draft must always exist. Deleting a published doc implies deleting
+        // the "previous" copy, since it only makes sense as a tool to revert
+        // the published doc's content. Note that deleting a draft recursively
+        // deletes both the published and previous docs.
+        async deleteOtherModes(req, doc, options) {
```

Comment:
"AfterDelete" was redundant, as handlers are not methods, they are namespaced to their event.